gist/src/main/java/build/gist/data/listeners/Queue.kt

```diff
@@ -124,6 +134,15 @@ class Queue: GistListener {
         }
     }
 
+    private fun addMessageToLocalStore(message: Message) {
+        val localMessage = localMessageStore.find { localMessage -> localMessage.queueId == message.queueId }
+        if (localMessage == null) { localMessageStore.add(message) }
+    }
+
+    private fun removeMessageFromLocalStore(message: Message) {
+        localMessageStore = localMessageStore.filter { localMessage -> localMessage.queueId != message.queueId } as MutableList<Message>
```

Comment:
What do you think about using the `removeAll{ it.queueId == message.queueId }` function instead of `filter`? 

Since `localMessageStore` is a mutable list, we can modify the object directly instead of re-assigning. 

The code would be identical, but I think `removeAll` would be more concise.